homeassistant/components/todoist/calendar.py

```diff
@@ -231,9 +232,10 @@ def _parse_due_date(data: dict, gmt_string) -> datetime:
     # Add time information to date only strings.
     if len(data["date"]) == 10:
         return datetime.fromisoformat(data["date"]).replace(tzinfo=dt.UTC)
-    if dt.parse_datetime(data["date"]).tzinfo is None:
+    nowTime = cast(datetime, dt.parse_datetime(data["date"]))
```

Comment:
Rather than hiding a potential problem with a cast, it might be better  to simply return `None` here if the date can't be parsed for some reason.  This is called in 2 places:
* [When calculating an end date](https://github.com/home-assistant/core/blob/116a9e351a2a8274fd41c2528e0967371442d7d6/homeassistant/components/todoist/calendar.py#L411-L413) - We could simply set the date date to `None` if we can't parse it.
* [When parsing the due date for an event](https://github.com/home-assistant/core/blob/116a9e351a2a8274fd41c2528e0967371442d7d6/homeassistant/components/todoist/calendar.py#L535-L537) - We could just `continue` here like it does above if there is no due date.

Reply:
that is a better solution, done.